vars/docOnlyChange.groovy

```diff
@@ -16,8 +16,19 @@ boolean call(String target_branch) {
     }
     if (cachedCommitPragma('Doc-only').toLowerCase() == 'false' ||
         !fileExists('ci/doc_only_change.sh')) {
-        return false
-    }
+        return sh(label: "Determine if doc-only change (manual mode with master branch)",
+                  script: ["TARGET_BRANCH=${target_branch}",
+                            '''
+                            set -uex
+                            if ! git fetch origin ${TARGET_BRANCH}; then
+                                echo "Hrm.  Got an error fetching the target branch"
+                                exit 0
+                            fi
+                            git diff --no-commit-id --name-only origin/${TARGET_BRANCH} HEAD |
+                               grep -v -e "^docs/" -e "\\.md$" -e "^.*LICENSE.*$"
+                            '''].join(' '),
```

Comment:
While a ` ` `join()`ed list works, we have typically just gone the route of a single string such as:

```suggestion
                  script: 'TARGET_BRANCH= ' + target_branch + '''
                            set -uex
                            if ! git fetch origin ${TARGET_BRANCH}; then
                                echo "Hrm.  Got an error fetching the target branch"
                                exit 0
                            fi
                            git diff --no-commit-id --name-only origin/${TARGET_BRANCH} HEAD |
                               grep -v -e "^docs/" -e "\\.md$" -e "^.*LICENSE.*$"
                            ''',
```

Although lines 34-36 are not lost on me.  I'm not entirely sure who wrote those and indeed, given that both of these if branches are duplicating the execution of an `sh()` step with the same arguments this whole function could be DRYed down into just setting the `script` variable inside the `if` and moving the duplicated parts after the `if` block.

But most importantly(!), adding the fall-back to the first branch of the `if` clause means needing to separate the or'd conditions in that branch as the `cachedCommitPragma('Doc-only').toLowerCase() == 'false'` is absolute and should not execute the script to determine if there are doc-only changes or not.

So something like:

```groovy
    if (cachedCommitPragma('Doc-only').toLowerCase() == 'false' {
        return false
    }

    if (fileExists('ci/doc_only_change.sh')) {
        script = 'CHANGE_ID= ' + env.CHANGE_ID +
                 ' TARGET_BRANCH= ' + target_branch + 
                 ' ci/doc_only_change.sh'
    } else {
        script = 'TARGET_BRANCH= ' + target_branch + '''
                  set -uex
                  if ! git fetch origin ${TARGET_BRANCH}; then
                      echo "Hrm.  Got an error fetching the target branch"
                      exit 0
                  fi
                  git diff --no-commit-id --name-only origin/${TARGET_BRANCH} HEAD |
                     grep -v -e "^docs/" -e "\\.md$" -e "^.*LICENSE.*$"'''
    }

    return sh(label: "Determine if doc-only change",
              script: script
              returnStatus: true) == 1
```

Reply:
Done